com.unity.render-pipelines.core/Editor/Material/MaterialHeaderScopeList.cs

```diff
@@ -27,9 +28,16 @@ public MaterialHeaderScopeList(uint defaultExpandedState = uint.MaxValue)
         /// <param name="title"><see cref="GUIContent"/> The title of the scope</param>
         /// <param name="expandable">The mask identifying the scope</param>
         /// <param name="action">The action that will be drawn if the scope is expanded</param>
-        public void RegisterHeaderScope(GUIContent title, uint expandable, Action<Material> action)
+        public void RegisterHeaderScope<TEnum>(GUIContent title, TEnum expandable, Action<Material> action)
```

Comment:
An enum might not necessarily be represented by an 4 byte variable. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum

Reply:
thanks for the info, I didn't think about this

Reply:
I looked into an issue around enums earlier today. Seems like Unity serialization does not support 8 byte enums, I am not sure about 2 bytes but I think that would not cause any problems with this.